File: languages/go/entity/identity/credential_offer_body.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type CredentialOfferBody struct {
	Url         string
	Credentials []CredentialOfferCredential
}

func NewCredentialOfferBodyFromProto(s *proto.CredentialOfferBody) CredentialOfferBody {
	if s == nil {
		return CredentialOfferBody{}
	}

	var credentials []CredentialOfferCredential
	for _, c := range s.Credentials {
		credentials = append(credentials, NewCredentialOfferCredentialsFromProto(c))
	}

	return CredentialOfferBody{
		Url:         s.Url,
		Credentials: credentials,
	}
}

func (c CredentialOfferBody) ToProto() *proto.CredentialOfferBody {
	var credentials []*proto.CredentialOfferBodyCredentials
	for _, a := range c.Credentials {
		credentials = append(credentials, a.ToProto())
	}

	return &proto.CredentialOfferBody{
		Url:         c.Url,
		Credentials: credentials,
	}
}

File: languages/js/src/entity/identity/credential_offer_body.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";
import { CredentialOfferBodyCredentials } from "./credential_offer_credentials";

export class CredentialOfferBody {
  url: string;
  credentials: CredentialOfferBodyCredentials[];

  constructor(url: string, credentials: CredentialOfferBodyCredentials[]) {
    this.url = url;
    this.credentials = credentials;
  }

  public toProto(): identityEntitiesProto.CredentialOfferBody {
    return identityEntitiesProto.CredentialOfferBody.fromPartial({
      url: this.url,
      credentials: this.credentials.map(c => c.toProto())
    });
  }

  static fromProto(
    r: identityEntitiesProto.CredentialOfferBody
  ): CredentialOfferBody {
    return new CredentialOfferBody(
      r.url,
      r.credentials.map(c => CredentialOfferBodyCredentials.fromProto(c))
    );
  }
}
